### PR TITLE
feat(DEV-1668): support auction house contracts

### PR DESCRIPTION
**Pull request by Superjo149**

### CODE REVIEW

File: src/solana/utils/utils.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export function toLamports(
   const amount =
     typeof account === "number" ? account : account.data.amount?.toNumber();
 
-  const precision = Math.pow(10, mint?.decimals || 0);
+  const precision = Math.pow(10, mint?.decimals || 9);
```
Comment (viet10101988) [inline src/solana/utils/utils.ts:26]: magic lol

File: tests/e2e/spl.e2e.spec.ts [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        await refinable.connection.getParsedAccountInfo(
+          toPublicKey(offer.blockchainId)
+        )
+      ).value
+    ).toBeNull();
+  });
+
+  it("Should be able to buy nft", async () => {
+    // Backend might not have processed the event
+    await sleep(2000);
```
Comment (viet10101988) [inline tests/e2e/spl.e2e.spec.ts:132]: 2s doesn't guarantee that event is processed?

File: tests/e2e/spl.e2e.spec.ts [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+    await tx.wait();
+
+    expect(
+      (
+        await refinable.connection.getParsedAccountInfo(
+          toPublicKey(offer.blockchainId)
+        )
+      ).value
+    ).toBeNull();
```
Comment (viet10101988) [inline tests/e2e/spl.e2e.spec.ts:185]: why is it null?

**Review by viet10101988 — CHANGES_REQUESTED** [review]

File: src/solana/ah.ts [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+  TOKEN_PROGRAM_ID,
+  u64,
+} from "@solana/spl-token";
+import { Connection, PublicKey } from "@solana/web3.js";
+import BN from "bn.js";
+import { Buffer } from "buffer";
+import { toPublicKey } from "./utils";
+
+const AUCTION_HOUSE_PROGRAM_ID = new PublicKey(
+  "hausS13jsjafwWwGqZTUQRmWyvyxn9EQpqMwV1PBBmk"
```
[the remark this answers is not in the document]
Comment (Superjo149) [reply src/solana/ah.ts:14]: No, should be coded inside the lib. This is pending to change as we might want to deploy our own program.

File: tests/e2e/spl.e2e.spec.ts (hunk 3 above)
Comment (Superjo149) [reply tests/e2e/spl.e2e.spec.ts:185]: Because the item has been bought

File: tests/e2e/spl.e2e.spec.ts (hunk 2 above)
Comment (Superjo149) [reply tests/e2e/spl.e2e.spec.ts:132]: Yes and no, just needs a bit more time, but not other easy way to check

File: src/refinable/RefinableSolanaClient.ts [hunk 5]
```diff
@@ -34,12 +32,7 @@ export class RefinableSolanaClient extends RefinableBaseClient<RefinableSolanaOp
     return refinable;
   }
 
-  async init() {
-    this.store = await Store.load(
-      this._connection,
-      solanaStorePubKeys[this.options.environment]
-    );
-  }
+  async init() {}
```
[the remark this answers is not in the document]
Comment (Superjo149) [reply src/refinable/RefinableSolanaClient.ts:35]: Yes

File: src/nft/tx.ts [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+            if (line.startsWith('Program log: ')) {
+              throw new Error(
+                'Transaction failed: ' + line.slice('Program log: '.length),
+              );
+            }
+          }
+        }
+        throw new Error(JSON.stringify(simulateResult.err));
+      }
+      // throw new Error('Transaction failed');
```
[the remark this answers is not in the document]
Comment (Superjo149) [reply src/nft/tx.ts:153]: Actually this file is no longer needed

File: src/nft/tx.ts [hunk 7]
```diff
@@ -0,0 +1,276 @@
...
+    TransactionSignature,
+  } from '@solana/web3.js';
+  
+  export const DEFAULT_TIMEOUT = 15000;
+
+  export const getUnixTs = () => {
+    return new Date().getTime() / 1000;
+  };
+  
+  export function sleep(ms: number): Promise<void> {
```
[the remark this answers is not in the document]
Comment (Superjo149) [reply src/nft/tx.ts:23]: Deleting the file as its no longer needed

**Review by viet10101988 — APPROVED** [review]

**Pull request merged**